addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1178,7 +1178,10 @@ const Wysiwyg = Widget.extend({
     _getSelectedColor($, eventName) {
         const selection = this.odooEditor.document.getSelection();
         const range = selection.rangeCount && selection.getRangeAt(0);
-        const targetNode = range && range.startContainer;
+        const targetNode = range && (range.startContainer.length === range.startOffset
+            ? range.endContainer
+            : range.startContainer
+        );
```

Comment:
Seems wrong... what about if I selected my text from right to left?

Reply:
Now using `nextLeaf(range.startContainer)` instead of `range.endContainer`.